test/gtest/common/test_helpers.cc

```diff
@@ -15,6 +15,134 @@ namespace ucs {
 
 const double test_timeout_in_sec = 60.;
 
+const double watchdog_timeout_default = 900.; // 15 minutes
+
+static test_watchdog_t watchdog;
+
+void *watchdog_func(void *arg)
+{
+    int ret = 0;
+    double now;
+    struct timespec timeout;
+
+    pthread_mutex_lock(&watchdog.mutex);
+
+    /* sync with parent thread */
+    pthread_barrier_wait(&watchdog.barrier);
+
+    do {
+        now = ucs_get_accurate_time();
+        ucs_sec_to_timespec(now + watchdog.timeout, &timeout);
+
+        ret = pthread_cond_timedwait(&watchdog.cv, &watchdog.mutex, &timeout);
+        if (!ret) {
+            switch (watchdog.state) {
+            case test_watchdog_t::WATCHDOG_RUN:
+                if (watchdog.timeout != watchdog_timeout_default) {
```

Comment:
do we really need this 'if'?

Reply:
this is to reset timeout that was set by a test via `watchdog_timeout_set()`.
we can implement `watchdog_timeout_reset()` to control it from a test, wdyt?

Reply:
we can assign unconditionally

Reply:
i got your idea. fixed